### ISSUE TITLE
gcc flags not appropriate

### ISSUE THREAD

**Issue by Tux** [Sereal/Sereal#254]
zstd/Makefile adds a truckload of `CFLAGS` that are only applicable if the compiler is GNU gcc, but cause epic failure when it is not, like when using HP C-ANSI-C on HP-UX.
Please guard that against non-gcc compilers. Our live is hard enough as it is.

**Comment by thomasseeling-ibm** [comment]
Please add a robust zstd/Makefile which works on AIX. I guess that's what Config.pm is intended for.
I tried `perl Makefile.PL CCFLAGS="$(perl -MConfig -e 'print $Config{ccflags}')"` but it didn't work.
Currently I have to patch out the CFLAGS definition when building.
The #pragma in `zstd.h` line 72 is also not suitable for IBM's xlC compiler.

```
*** zstd/Makefile       Wed Jul  8 13:59:00 2026
--- zstd/Makefile.b     Wed Jul  8 13:43:59 2026
***************
*** 21,29 ****
  CC       = xlc_r
  CPPFLAGS+= -I. -I./common -DXXH_NAMESPACE=ZSTD_
  CFLAGS  ?= -O3
! CFLAGS  += -fPIC -Wall -Wextra -Wcast-qual -Wcast-align -Wshadow -Wstrict-aliasing=1 \
!            -Wswitch-enum -Wdeclaration-after-statement -Wstrict-prototypes -Wundef \
!            -Wpointer-arith
  CFLAGS  += $(MOREFLAGS)
  AR       = ar
  ARFLAGS  = rcs
--- 21,27 ----
  CC       = xlc_r
  CPPFLAGS+= -I. -I./common -DXXH_NAMESPACE=ZSTD_
  CFLAGS  ?= -O3
! CFLAGS  += -fPIC
  CFLAGS  += $(MOREFLAGS)
  AR       = ar
  ARFLAGS  = rcs
```

**Comment by Tux** [comment]
Yes, use `%Config` from `use Config;`. Your suggestion won't fly on HP-UX HP C-ANSI-C compiler, which doesn't know about `-fPIC`

**Comment by demerphq** [comment]
Fixed in 5.007, thanks for the report.

**Comment by thomasseeling-ibm** [comment]
> Yes, use `%Config` from `use Config;`. Your suggestion won't fly on HP-UX HP C-ANSI-C compiler, which doesn't know about `-fPIC`

Sorry, my shop only uses AIX and Linux :-)
